### PR TITLE
Integration with SSLService

### PR DESCRIPTION
**Pull request by gtaban**
Integrating SSLService into Kitura-net.

## Description

3 file changes:
1 - Package.swift: we need both new versions of BlueSocket and SSLService.
2 - HTTPServer.swift: add the delegate to the socket.
3 - IncomingSocketHandler.swift: In handleRead() make sure to close the connection incase of error in SSLservice or Socket.

## Checklist:

- [ ] I have submitted a [CLA form](https://github.com/IBM-Swift/CLA)
- [ ] If applicable, I have updated the documentation accordingly.
- [ ] If applicable, I have added tests to cover my changes.

### CODE REVIEW

File: Sources/KituraNet/HTTP/HTTPServer.swift [hunk 1]
```diff
@@ -51,19 +59,35 @@ public class HTTPServer {
...
+
+            // If SSL config has been created, 
+            // create and attach the SSLService delegate to the socket
+            if let sslConfig = sslConfig {
+                self.listenSocket?.delegate = try SSLService(usingConfiguration: sslConfig);
+            }
+            
+        } catch let error {
+            
+            if error is Socket.Error {
```
Comment (ianpartridge) [inline Sources/KituraNet/HTTP/HTTPServer.swift]: The best practice here is something like:

```swift
if let socketError = error as? Socket.Error {
    Log.error("Error reported:\n \(socketError.description)")
} else if let sslError = error as? SSLError {
    Log.error("Error reported:\n \(sslError.description)")
} ...
```

File: Sources/KituraNet/HTTP/HTTPServer.swift [hunk 2]
```diff
@@ -51,19 +59,35 @@ public class HTTPServer {
     /// - Parameter errorHandler: optional callback for error handling
     public func listen(port: Int, errorHandler: ((Swift.Error) -> Void)? = nil) {
         self.port = port
+
         do {
-            self.listenSocket = try Socket.create()
-        } catch {
-            if let callback = errorHandler {
-                callback(error)
+			self.listenSocket = try Socket.create()
```
Comment (ianpartridge) [inline Sources/KituraNet/HTTP/HTTPServer.swift]: Spaces not tabs.

File: Sources/KituraNet/HTTP/HTTPServer.swift [hunk 3]
```diff
@@ -51,19 +59,35 @@ public class HTTPServer {
...
+                
             } else {
                 Log.error("Error creating socket: \(error)")
             }
         }
 
         guard let socket = self.listenSocket else {
-            // already did a callback on the error handler or logged error
-            return
+        // already did a callback on the error handler or logged error
```
Comment (ianpartridge) [inline Sources/KituraNet/HTTP/HTTPServer.swift]: Should still be indented.

File: Sources/KituraNet/HTTP/HTTPServer.swift [hunk 4]
```diff
@@ -18,15 +18,23 @@ import Dispatch
...
 
 // MARK: HTTPServer
 
 /// An HTTP server that listens for connections on a socket.
 public class HTTPServer {
 
-    /// HTTP `ServerDelegate`.
+    ///
+    /// HTTPServerDelegate
+    ///
```
Comment (ianpartridge) [inline Sources/KituraNet/HTTP/HTTPServer.swift]: I don't think this comment should have been changed?

File: Sources/KituraNet/HTTP/HTTPServer.swift [hunk 5]
```diff
@@ -18,15 +18,23 @@ import Dispatch
...
     public weak var delegate: ServerDelegate?
     
+    ///
+    /// SSL cert configs for handling client requests
+    ///
```
Comment (ianpartridge) [inline Sources/KituraNet/HTTP/HTTPServer.swift]: One-line comment only, please.

**Comment by gtaban** [comment]
ok made the changes. Let me know.

File: Sources/KituraNet/HTTP/HTTPServer.swift (hunk 4 above)
Comment (gtaban) [inline Sources/KituraNet/HTTP/HTTPServer.swift]: You're right. Fixed.

File: Sources/KituraNet/HTTP/HTTPServer.swift (hunk 5 above)
Comment (gtaban) [inline Sources/KituraNet/HTTP/HTTPServer.swift]: Done.

File: Sources/KituraNet/HTTP/HTTPServer.swift (hunk 1 above)
Comment (gtaban) [inline Sources/KituraNet/HTTP/HTTPServer.swift]: done.

**Pull request merged**